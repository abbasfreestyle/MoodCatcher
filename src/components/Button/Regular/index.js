import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';

import Touchable from 'components/Touchable';
import Grid from 'components/Grid';

const styles = StyleSheet.create({
  // eslint-disable-next-line
  container: {
    backgroundColor: 'purple',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10
  },
  // eslint-disable-next-line
  text: {
    color: 'white',
    fontSize: 18,
    fontWeight: '400',
    textTransform: 'uppercase'
  }
});

class Regular extends PureComponent {
  render() {
    const { children, onPress, flex, margin } = this.props;
    return (
      <Grid.Row margin={margin}>
        <View flex={flex ? 1 : 0}>
          <Touchable onPress={onPress} borderRadius={5}>
            <View style={styles.container}>
              <Text style={styles.text}>{children}</Text>
            </View>
          </Touchable>
        </View>
      </Grid.Row>
    );
  }
}

Regular.defaultProps = {
  onPress: () => {},
  flex: false,
  margin: 0
};

Regular.propTypes = {
  children: PropTypes.string.isRequired,
  flex: PropTypes.bool,
  margin: PropTypes.number,
  onPress: PropTypes.func
};

export default Regular;
