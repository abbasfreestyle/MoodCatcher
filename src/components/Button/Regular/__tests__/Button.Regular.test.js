import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Regular from '..';

const requiredProps = {
  children: 'Happy',
  onPress: jest.fn()
};

jest.useFakeTimers();

const wrapper = shallow(<Regular {...requiredProps} />);

describe('Button > Regular.', () => {
  describe('Snapshots.', () => {
    afterEach(() => {
      wrapper.setProps({ ...Regular.defaultProps, ...requiredProps });
      jest.advanceTimersByTime(0);
    });

    test('should match default props', () => {
      expect(Regular.defaultProps).toMatchSnapshot();
    });

    test('should match with default props provided', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when disabled is true', () => {
      wrapper.setProps({ disabled: true });
      jest.advanceTimersByTime(200);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when flex is true', () => {
      wrapper.setProps({ flex: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when margin is different', () => {
      wrapper.setProps({ margin: 20 });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
