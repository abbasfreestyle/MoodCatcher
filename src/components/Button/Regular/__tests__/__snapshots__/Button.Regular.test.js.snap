// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button > Regular. Snapshots. should match default props 1`] = `
Object {
  "disabled": false,
  "flex": false,
  "margin": 0,
}
`;

exports[`Button > Regular. Snapshots. should match when disabled is true 1`] = `
<Row
  margin={0}
>
  <View
    flex={0}
  >
    <Touchable
      borderRadius={5}
      disabled={true}
      onPress={[MockFunction]}
      rippleColor="#FFFFFF"
    >
      <AnimatedComponent
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(238, 238, 238, 1)",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "400",
              "textTransform": "uppercase",
            }
          }
        >
          Happy
        </Text>
      </AnimatedComponent>
    </Touchable>
  </View>
</Row>
`;

exports[`Button > Regular. Snapshots. should match when flex is true 1`] = `
<Row
  margin={0}
>
  <View
    flex={1}
  >
    <Touchable
      borderRadius={5}
      disabled={false}
      onPress={[MockFunction]}
      rippleColor="#FFFFFF"
    >
      <AnimatedComponent
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(68, 187, 164, 1)",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "400",
              "textTransform": "uppercase",
            }
          }
        >
          Happy
        </Text>
      </AnimatedComponent>
    </Touchable>
  </View>
</Row>
`;

exports[`Button > Regular. Snapshots. should match when margin is different 1`] = `
<Row
  margin={20}
>
  <View
    flex={0}
  >
    <Touchable
      borderRadius={5}
      disabled={false}
      onPress={[MockFunction]}
      rippleColor="#FFFFFF"
    >
      <AnimatedComponent
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(68, 187, 164, 1)",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "400",
              "textTransform": "uppercase",
            }
          }
        >
          Happy
        </Text>
      </AnimatedComponent>
    </Touchable>
  </View>
</Row>
`;

exports[`Button > Regular. Snapshots. should match with default props provided 1`] = `
<Row
  margin={0}
>
  <View
    flex={0}
  >
    <Touchable
      borderRadius={5}
      disabled={false}
      onPress={[MockFunction]}
      rippleColor="#FFFFFF"
    >
      <AnimatedComponent
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(68, 187, 164, 1)",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "400",
              "textTransform": "uppercase",
            }
          }
        >
          Happy
        </Text>
      </AnimatedComponent>
    </Touchable>
  </View>
</Row>
`;
